packages/next/server/render.tsx

```diff
@@ -1807,15 +1798,9 @@ function renderToStream({
           reject(err)
         }
       },
-      onCompleteShell() {
-        if (!generateStaticHTML) {
-          doResolve()
-        }
-      },
-      onCompleteAll() {
-        doResolve()
-      },
```

Comment:
We don't want to remove this, because otherwise we will serve partial streams to crawlers. Instead, you can make the function `async` now (instead of returning a promise) and do something like this instead:

```tsx
let resolveAllCompleted
const allCompleted = new Promise(resolve => { resolveAllCompleted = resolve })
const renderStream = await ReactDOMServer.renderToReadableStream(element, {
  onError(err: Error) {
    // ...
  },
  onCompleteAll() {
    resolveAllCompleted()
  }
}

if (generateStaticHTML) {
  // Wait for the entire stream to complete before resolving
  await allCompleted
}

doResolve(renderStream)
```

Reply:
There should be a test case against this, we should also find out why it didn't fail here...